zaza/openstack/charm_tests/designate/tests.py

```diff
@@ -0,0 +1,491 @@
+        logging.debug('Checking designate:designate-bind dns-backend relation'
+                      'data...')
+
+        unit_name = 'designate/0'
+        remote_unit_name = 'designate-bind/0'
+        relation_name = 'dns-backend'
+        remote_unit = model.get_unit_from_name(remote_unit_name)
+        remote_ip = remote_unit.public_address
+        relation = zaza_juju.get_relation_from_unit(
+            unit_name,
+            remote_unit_name,
+            relation_name
+        )
+        # Get private-address in relation
+        rel_private_ip = relation.get('private-address')
+        # The private address in relation should match designate-bind/0 address
+        self.assertEqual(rel_private_ip, remote_ip)
+
+    def test_206_designate_bind_designate_relation(self):
+        """Verify the designate_bind to designate dns-backend relation data."""
+        logging.debug('Checking designate-bind:designate dns-backend relation'
+                      'data...')
+
+        unit_name = 'designate-bind/0'
+        remote_unit_name = 'designate/0'
+        relation_name = 'dns-backend'
+        remote_unit = model.get_unit_from_name(remote_unit_name)
+        remote_ip = remote_unit.public_address
+        relation = zaza_juju.get_relation_from_unit(
+            unit_name,
+            remote_unit_name,
+            relation_name
+        )
+        # Get private-address in relation
+        rel_private_ip = relation.get('private-address')
+        # The private address in relation should match designate-bind/0 address
+        self.assertEqual(rel_private_ip, remote_ip)
+
+
+class ResetAndPauseTests(BaseDesignateTest):
+    """Designate charm restart and pause tests."""
+
+    def test_900_restart_on_config_change(self):
+        """Checking restart happens on config change.
+
+        Change disk format and assert then change propagates to the correct
+        file and that services are restarted as a result
+        """
+        # Expected default and alternate values
+        set_default = {'debug': 'False'}
+        set_alternate = {'debug': 'True'}
+
+        # Services which are expected to restart upon config change,
+        # and corresponding config files affected by the change
+        conf_file = '/etc/designate/designate.conf'
+
+        # Make config change, check for service restarts
+        self.restart_on_changed(
+            conf_file,
+            set_default,
+            set_alternate,
+            {'DEFAULT': {'debug': ['False']}},
+            {'DEFAULT': {'debug': ['True']}},
+            self.designate_svcs)
+
+    def test_910_pause_and_resume(self):
+        """Run pause and resume tests.
+
+        Pause service and check services are stopped then resume and check
+        they are started
+        """
+        with self.pause_resume(
+                self.designate_svcs,
+                pgrep_full=False):
+            logging.info("Testing pause resume")
+
+
+class ServerCreationTest(BaseDesignateTest):
```

Comment:
I think we need one overarching class DesignateTests which includes ServerCreationTest and DomainCreationTests and the restart and puase/resume tests.